src/algebra/algebra/tower.lean

```diff
@@ -0,0 +1,276 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import algebra.algebra.subalgebra
+
+/-!
+# Towers of algebras
+
+In this file we prove basic facts about towers of algebra.
+
+An algebra tower A/S/R is expressed by having instances of `algebra A S`,
+`algebra R S`, `algebra R A` and `is_scalar_tower R S A`, the later asserting the
+compatibility condition `(r • s) • a = r • (s • a)`.
+
+An important definition is `to_alg_hom R S A`, the canonical `R`-algebra homomorphism `S →ₐ[R] A`.
+
+-/
+
+universes u v w u₁ v₁
+
+variables (R : Type u) (S : Type v) (A : Type w) (B : Type u₁) (M : Type v₁)
+
+namespace algebra
+
+variables [comm_semiring R] [semiring A] [algebra R A]
+variables [add_comm_monoid M] [semimodule R M] [semimodule A M] [is_scalar_tower R A M]
+
+variables {A}
+
+/-- The `R`-algebra morphism `A → End (M)` corresponding to the `A`-module structure on `M`. -/
```

Comment:
I wonder if it would be worth including the keyword "representation" in this docstring. After all this is really the data of a representation of the algebra `A` on the module `M`.

Reply:
Where should I include the keyword?

Reply:
```suggestion
/-- The `R`-algebra morphism `A → End (M)` corresponding to the representation of the algebra `A` on the `R`-module `M`. -/
```